From a0ea332df477480f90d9dc841faf71f4b5be892e Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Wed, 29 Nov 2023 12:45:59 +0530
Subject: [PATCH 2/4] Change the working directory to source to workaround
 cython embedding absolute path to the pyx file in output.

Upstream-Status: Inappropriate [OE build specific]

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
---
 .../python/src/compatibility/openvino/cmake/UseCython.cmake     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bindings/python/src/compatibility/openvino/cmake/UseCython.cmake b/src/bindings/python/src/compatibility/openvino/cmake/UseCython.cmake
index 03a208f03c2..be9af591b34 100644
--- a/src/bindings/python/src/compatibility/openvino/cmake/UseCython.cmake
+++ b/src/bindings/python/src/compatibility/openvino/cmake/UseCython.cmake
@@ -258,7 +258,7 @@ function( compile_pyx _name generated_file )
   add_custom_command( OUTPUT ${_generated_file}
     COMMAND ${CYTHON_EXECUTABLE}
     ARGS ${cxx_arg} ${include_directory_arg} ${version_arg}
-    ${annotate_arg} ${no_docstrings_arg} ${cython_debug_arg} ${CYTHON_FLAGS}
+    ${annotate_arg} ${no_docstrings_arg} ${cython_debug_arg} ${CYTHON_FLAGS} -w ${CMAKE_CURRENT_SOURCE_DIR}
     --output-file  ${_generated_file} ${pyx_locations}
     DEPENDS ${pyx_locations} ${pxd_dependencies} ${pxi_dependencies}
     IMPLICIT_DEPENDS ${pyx_lang} ${c_header_dependencies}
-- 
2.34.1

